keylkgSup/keylkg.db

```diff
@@ -0,0 +1,454 @@
+record(bo, "$(P)SIM")
+{
+    field(SCAN, "Passive")
+    field(DTYP, "Soft Channel")
+    field(ZNAM, "NO")
+    field(ONAM, "YES")
+    field(VAL, "$(RECSIM=0)")
+    field(PINI, "YES")
+}
+
+record(bo, "$(P)DISABLE")
+{
+    field(DESC, "Disable comms")
+    field(PINI, "YES")
+    field(VAL, "$(DISABLE=0)")
+    field(OMSL, "supervisory")
+    field(ZNAM, "COMMS ENABLED")
+    field(ONAM, "COMMS DISABLED")
+}
+
+
+# The device has a set of commands that are available in its normal mode
+# and its measurement mode. These disable records are used to control what's
+# enabled and disabled depending on the mode. The DISABLE record supersedes
+# both of these however. 
+record(calc, "$(P)DISABLE_IF_COMM_MODE")
+{
+    field(DESC, "Disable record for communication mode")
+    field(INPA, "$(P)DISABLE CP")
+    field(INPB, "$(P)MODE CP")
+    
+    field(CALC, "(A=1)OR(B=1)?1:0")
+    field(ASG, "READONLY")
+}
+
+record(calc, "$(P)DISABLE_IF_NORM_MODE")
+{
+    field(DESC, "Disable record for normal mode")
+    field(INPA, "$(P)DISABLE CP")
+    field(INPB, "$(P)MODE CP")
+    
+    field(CALC, "(A=1)OR(B=0)?1:0")
```

Comment:
The mode PV will get reset to zero whenever the IOC restarts. This will then cause `DISABLE_IF_NORM_MODE` to be True, which will stop various records from reading from the device. This is likely not the desired behaviour: if the IOC was restarted in the middle of the night for whatever reason, they would lose the log data until someone noticed the measurements were no longer updating.

We need the IOC to shout very loudly about this if it occurs - e.g. the reads to go into a major alarm and big red text on the OPI.

Reply:
Added alarm state for `SET_UP` mode and a dynamic warning message for the user exists on the OPI when switching between modes.